core/src/main/java/org/elasticsearch/threadpool/ThreadPool.java

```diff
@@ -222,7 +223,8 @@ public ThreadPool(Settings settings) {
         int halfProcMaxAt5 = Math.min(((availableProcessors + 1) / 2), 5);
         int halfProcMaxAt10 = Math.min(((availableProcessors + 1) / 2), 10);
         Map<String, Settings> defaultExecutorTypeSettings = new HashMap<>();
-        add(defaultExecutorTypeSettings, new ExecutorSettingsBuilder(Names.GENERIC).size(4 * availableProcessors).keepAlive("30s"));
+        int genericThreadPoolSize = Math.min(512, Math.max(128, 4 * availableProcessors));
```

Comment:
You know it might make sense to make a `boundedBy` method that takes the calculated value, the min value, and the max value. It can be implemented just like this but then it is super obvious what it does.